src/stepfunctions/steps/sagemaker.py

```diff
@@ -518,22 +526,26 @@ def __init__(self, state_id, processor, job_name, inputs=None, outputs=None, exp
                                                                        SageMakerApi.CreateProcessingJob)
 
         if isinstance(job_name, str):
-            parameters = processing_config(processor=processor, inputs=inputs, outputs=outputs, container_arguments=container_arguments, container_entrypoint=container_entrypoint, kms_key_id=kms_key_id, job_name=job_name)
+            processing_parameters = processing_config(processor=processor, inputs=inputs, outputs=outputs, container_arguments=container_arguments, container_entrypoint=container_entrypoint, kms_key_id=kms_key_id, job_name=job_name)
         else:
-            parameters = processing_config(processor=processor, inputs=inputs, outputs=outputs, container_arguments=container_arguments, container_entrypoint=container_entrypoint, kms_key_id=kms_key_id)
+            processing_parameters = processing_config(processor=processor, inputs=inputs, outputs=outputs, container_arguments=container_arguments, container_entrypoint=container_entrypoint, kms_key_id=kms_key_id)
 
         if isinstance(job_name, Placeholder):
```

Comment:
thought: I was surprised we don't need to something like like this for `container_arguments`, `kms_key_id`, and `container_entrypoint` since [`processing_config` accepts `[str]` or `str` for these fields](https://github.com/aws/sagemaker-python-sdk/blob/master/src/sagemaker/workflow/airflow.py#L1108-L1110). However digging into the implementation, the values are just assigned to the corresponding CreateProcessingJob API field without any type checks.

https://github.com/aws/sagemaker-python-sdk/blob/467390f79e84310e0f987b3c19c095941b6925bd/src/sagemaker/processing.py#L961,L963

https://github.com/aws/sagemaker-python-sdk/blob/467390f79e84310e0f987b3c19c095941b6925bd/src/sagemaker/processing.py#L981-L982

Reply:
Had the same thought! :)